simple_switch_grpc not compiling

I an trying to compile the simple_switch_grpc with thrift support. I run the following commands but make fails. Iam using the latest behavioral-model code on github.

`~/behavioral-model/targets/simple_switch_grpc$ ./configure 'CXXFLAGS=-O0 -g' --with-thrift --with-grpc --with-pi`

The error is :
```
~/behavioral-model/targets/simple_switch_grpc$ make 
make  all-recursive
make[1]: Entering directory '/home/simo/behavioral-model/targets/simple_switch_grpc'
Making all in .
make[2]: Entering directory '/home/simo/behavioral-model/targets/simple_switch_grpc'
depbase=`echo switch_runner.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.  -I./../../include -I./../simple_switch -I./../../PI -isystem./../../third_party/spdlog  -DWITH_THRIFT -I./../simple_switch/gen-cpp -Icpp_out -Igrpc_out -I./cpp_out -I./grpc_out  -Wall -Wextra -pthread -g -O2 -std=c++11 -MT switch_runner.lo -MD -MP -MF $depbase.Tpo -c -o switch_runner.lo switch_runner.cpp &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I./../../include -I./../simple_switch -I./../../PI -isystem./../../third_party/spdlog -DWITH_THRIFT -I./../simple_switch/gen-cpp -Icpp_out -Igrpc_out -I./cpp_out -I./grpc_out -Wall -Wextra -pthread -g -O2 -std=c++11 -MT switch_runner.lo -MD -MP -MF .deps/switch_runner.Tpo -c switch_runner.cpp  -fPIC -DPIC -o .libs/switch_runner.o
In file included from switch_runner.cpp:49:0:
switch_runner.h:33:7: error: using typedef-name ‘grpc::Server’ after ‘class’
 class Server;
       ^
In file included from /usr/local/include/grpcpp/grpcpp.h:58:0,
                 from /usr/local/include/grpc++/grpc++.h:26,
                 from switch_runner.cpp:37:
/usr/local/include/grpcpp/server.h:26:29: note: ‘grpc::Server’ has a previous declaration here
 typedef ::grpc_impl::Server Server;
                             ^
Makefile:792: recipe for target 'switch_runner.lo' failed
make[2]: *** [switch_runner.lo] Error 1
make[2]: Leaving directory '/home/simo/behavioral-model/targets/simple_switch_grpc'
Makefile:886: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/simo/behavioral-model/targets/simple_switch_grpc'
Makefile:545: recipe for target 'all' failed
make: *** [all] Error 2
```

I tried with different verions , 1.17 , 1.18 , 1.20, and 1.30.x, they all gave linking errors (undefined reference to grpc_impl::..) when I compile the simple_switch_grpc except the last version (1.30.x) it gives that compilation error I showed above.

--- a/targets/simple_switch_grpc/switch_runner.cpp
+++ b/targets/simple_switch_grpc/switch_runner.cpp
@@ -34,7 +34,7 @@
 #include <PI/target/pi_imp.h>
 #include <PI/target/pi_learn_imp.h>
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 

--- a/targets/simple_switch_grpc/switch_runner.h
+++ b/targets/simple_switch_grpc/switch_runner.h
@@ -23,16 +23,12 @@
 
 #include <bm/bm_sim/dev_mgr.h>
 
+#include <grpcpp/server.h>
+
 #include <memory>
 #include <string>
 
 class SimpleSwitch;
-
-namespace grpc {
-
-class Server;
-
-}  // namespace grpc
 
 namespace bm {
 

--- a/targets/simple_switch_grpc/tests/base_test.cpp
+++ b/targets/simple_switch_grpc/tests/base_test.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <google/rpc/code.pb.h>
 

--- a/targets/simple_switch_grpc/tests/example.cpp
+++ b/targets/simple_switch_grpc/tests/example.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <google/rpc/code.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_action_profile.cpp
+++ b/targets/simple_switch_grpc/tests/test_action_profile.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_basic.cpp
+++ b/targets/simple_switch_grpc/tests/test_basic.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/v1/p4runtime.grpc.pb.h>
 

--- a/targets/simple_switch_grpc/tests/test_counter.cpp
+++ b/targets/simple_switch_grpc/tests/test_counter.cpp
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_digest.cpp
+++ b/targets/simple_switch_grpc/tests/test_digest.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_gnmi.cpp
+++ b/targets/simple_switch_grpc/tests/test_gnmi.cpp
@@ -20,7 +20,7 @@
 
 #include <bm/bm_sim/logger.h>
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <gnmi/gnmi.grpc.pb.h>
 #include <p4/bm/dataplane_interface.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_grpc_dp.cpp
+++ b/targets/simple_switch_grpc/tests/test_grpc_dp.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 

--- a/targets/simple_switch_grpc/tests/test_idle_timeout.cpp
+++ b/targets/simple_switch_grpc/tests/test_idle_timeout.cpp
@@ -19,7 +19,7 @@
  */
 
 #include <google/protobuf/util/message_differencer.h>
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_meter.cpp
+++ b/targets/simple_switch_grpc/tests/test_meter.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_optional.cpp
+++ b/targets/simple_switch_grpc/tests/test_optional.cpp
@@ -21,7 +21,7 @@
 // TODO(antonin): we should try to unify this test suite with the "ternary" one,
 // as pretty much all of the code is exactly the same.
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_packet_io.cpp
+++ b/targets/simple_switch_grpc/tests/test_packet_io.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/v1/p4runtime.grpc.pb.h>
 #include <PI/proto/pi_server.h>

--- a/targets/simple_switch_grpc/tests/test_pre.cpp
+++ b/targets/simple_switch_grpc/tests/test_pre.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <google/rpc/status.pb.h>
 #include <p4/bm/dataplane_interface.grpc.pb.h>

--- a/targets/simple_switch_grpc/tests/test_ternary.cpp
+++ b/targets/simple_switch_grpc/tests/test_ternary.cpp
@@ -18,7 +18,7 @@
  *
  */
 
-#include <grpc++/grpc++.h>
+#include <grpcpp/grpcpp.h>
 
 #include <p4/bm/dataplane_interface.grpc.pb.h>
 #include <p4/v1/p4runtime.grpc.pb.h>

